python/rapidsmpf/rapidsmpf/integrations/dask/shuffler.py

```diff
@@ -282,11 +316,16 @@ def _extract_partition(
     """
     if callback is None:
         raise ValueError("Missing callback in _extract_partition.")
-    return callback(
+    shuffler = get_shuffler(shuffle_id)
+    ret = callback(
```

Comment:
What do we want to do if `callback` errors? Maybe this should instead be

```python
try:
    ret = callback(...)
finally:
    ctx = get_worker_context()
    del ctx.shufflers[shuffle_id]
else:
    return ret
```

Reply:
The problem is that to call `shuffler.finished()`, we need to call `get_shuffler()`, which might create a new `Shuffler`.

I don't think we should guard against calling `_extract_partition` with the wrong types. But it is a good idea to guard against errors in `callback`. It now looks like:
```python
    shuffler = get_shuffler(shuffle_id)
    try:
        return callback(
            partition_id,
            column_names,
            shuffler,
        )
    finally:
        if shuffler.finished():
            ctx = get_worker_context()
            del ctx.shufflers[shuffle_id]
```